### ISSUE TITLE
fluid的op精度比pytorch低

### ISSUE THREAD

**Issue by superwj1990** [PaddlePaddle/Paddle#16263]
经实验发现，fluid的op精度低于pytorch，在求loss时会导致收敛速度变慢。需要在softmax之前cast成float64解决。

**Comment by superwj1990** [comment]
在识别服务中，基于相同的数据库，采用resnet34结构。对比了：全连接层（act：softmax）+cross_entroy和全连接层（cast to float64）+softmax_with_cross_entropy两种loss计算方式。前者第8个epoch的acc约等于5%，后者第2个epoch的acc约等于40%，可见精度已经严重影响了收敛速度。

**Comment by qingqing01** [comment]
- FP32测试:  全连接层（act: softmax）+ cross_entroy  和  softmax_with_cross_entropy两种loss计算方式，这个在ImageNet分类+MobileNet等模型上验证过一致 。

- FP64的cast，能贴下复现方式吗? @superwj1990

**Comment by shippingwang** [comment]
FP32测试
ResNet 50 
softmax + cross_entropy
![image](https://user-images.githubusercontent.com/13645332/54587828-30f01d80-4a5c-11e9-93b0-9e7f7e858aa3.png)

softmax_with_cross_entropy
![image](https://user-images.githubusercontent.com/13645332/54587849-406f6680-4a5c-11e9-910c-b4eeb944b2aa.png)

验证结果一致

**Comment by shippingwang** [comment]
@superwj1990 麻烦贴一下 cast64的相关代码 我在fc前 fc后加cast 64 都报错
